[Inductor] [CI] [CUDA] Skip the failed models and tests the better way

nWEIdia opened the pull request:
Address subtasks in https://github.com/pytorch/pytorch/issues/126692 

After enabling the disabled shards, the following two models regressed (for cu124 configuration): 
dynamic_inductor_timm_training.csv  
cspdarknet53,pass,7   (expected)                                        | cspdarknet53,fail_accuracy,7           (actual)
eca_botnext26ts_256,pass,7        (expected)                            | eca_botnext26ts_256,fail_accuracy,7 (actual)

cc @albanD @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @atalman @malfet @ptrblck @eqy @tinglvv @Aidyn-A

nWEIdia committed:
Revert "Disabling test shards given https://github.com/pytorch/pytorch/issues/126692"
Add back shards and disable models by setting the accuracy to "fail
Relax inductor perf smoke speedup from 4.9 to 4.7 for cuda 12.4 only
Duplicate cu121 csv reference to cu124 to workaround https://github.com/pytorch/pytorch/issues/126692

Change back to pass after fixing https://github.com/pytorch/pytorch/issues/126692

huydhn commented:
@nWEIdia I should have pinged you earlier, but I have a small fix for a 12.4/12.1 typo on inductor workflow here https://github.com/pytorch/pytorch/pull/127121 that has just been landed.  You might want to merge in that change.

huydhn commented on benchmarks/dynamo/ci_expected_accuracy/dynamic_inductor_timm_training.csv:
```diff
@@ -6,7 +6,7 @@ adv_inception_v3,pass,6
 
 
-beit_base_patch16_224,pass,7
+beit_base_patch16_224,fail_accuracy,7
```
I realize that there is no clear way to tell tat this is failing accuracy only on 12.4?

nWEIdia replied:
You are right. I am still trying to figure out how to branch as CUDA 12.1 would pass...

nWEIdia replied:
Duplicated the csv files to a cu124 directory and it seems to work.

nWEIdia commented:
> @nWEIdia I should have pinged you earlier but missed your PR. I have a small fix for a 12.4/12.1 typo on inductor workflow here #127121 that has just been landed. You might want to merge in that change.

Thank you for the fix!

nWEIdia commented on .ci/pytorch/test.sh:
```diff
@@ -589,7 +589,8 @@ test_inductor_torchbench_smoketest_perf() {
   # we switch to use some other model.
-  python benchmarks/dynamo/check_perf_csv.py -f "$TEST_REPORTS_DIR/inductor_inference_smoketest.csv" -t 4.9
+  # Change 4.9 to 4.7, change back after fixing https://github.com/pytorch/pytorch/issues/126692
+  python benchmarks/dynamo/check_perf_csv.py -f "$TEST_REPORTS_DIR/inductor_inference_smoketest.csv" -t 4.7
```
And here we probably need to if else for cuda 12.1 and cuda 12.4 too..

nWEIdia replied:
Did if else switch to use 4.9 for cu121 and 4.7 for cu124

nWEIdia committed:
Trying to fix torch.version error.

nWEIdia committed:
Revert "Trying to fix torch.version error."

This reverts commit 14a1867eb963ea0508758baeddec28dc5cff4dd7.

nWEIdia committed:
Reverting previous complicated fix. Retry pushd/popd
Just do not import torch under torch/ directory

nWEIdia committed:
Try torch.version.cuda in a different location

nWEIdia committed:
// in the path does not work, how about /./?

nWEIdia committed:
Too many typos

nWEIdia committed:
Avoid importing torch in *bazel* jobs
(https://github.com/pytorch/pytorch/actions/runs/9297551188/job/25587899137,
https://github.com/pytorch/pytorch/actions/runs/9297551188/job/25587898918,
https://github.com/pytorch/pytorch/actions/runs/9297551188/job/25587898648)

Add two more test failures skip for cu124...

nWEIdia committed:
Tweak model pass/fail_accuracy status

atalman reviewed (approved):
lgtm

nWEIdia commented:
@pytorchbot merge
